File: app/src/main/java/kztproject/jp/splacounter/MyApplication.java
package kztproject.jp.splacounter;

import android.app.Application;

import com.facebook.stetho.Stetho;

import javax.inject.Singleton;

import dagger.Component;
import kztproject.jp.splacounter.di.AppComponent;
import kztproject.jp.splacounter.di.AppModule;
import kztproject.jp.splacounter.preference.PrefsWrapper;

public class MyApplication extends Application {

    public static final String PREF = "pref";

    @Singleton
    @Component(modules = AppModule.class)
    public interface AppAppComponent extends AppComponent {
    }

    private AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        if (mAppComponent == null) {
            mAppComponent = DaggerMyApplication_AppAppComponent.builder()
                    .appModule(new AppModule(this))
                    .build();
        }

        if (BuildConfig.DEBUG) {
            Stetho.initializeWithDefaults(this);
        }

        PrefsWrapper.INSTANCE.initialize(getApplicationContext());
     }

    public AppComponent component() {
        return mAppComponent;
    }

    public void setmAppComponent(AppComponent component) {
        mAppComponent = component;
    }
}

File: app/src/main/java/kztproject/jp/splacounter/MyApplication.kt
package kztproject.jp.splacounter

import android.app.Activity
import android.app.Application
import com.facebook.stetho.Stetho
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import kztproject.jp.splacounter.di.AppComponent
import kztproject.jp.splacounter.di.DaggerAppComponent
import javax.inject.Inject

class MyApplication : Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    private lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        DaggerAppComponent.builder()
                .application(this)
                .build()
                .inject(this)

        if (BuildConfig.DEBUG) {
            Stetho.initializeWithDefaults(this)
        }
    }

    override fun activityInjector(): AndroidInjector<Activity> {
        return dispatchingAndroidInjector
    }

    fun component(): AppComponent {
        return appComponent
    }
}
